executor/envd/server.py

```
@@ -323,14 +345,23 @@ async def _handle_unary(request: Request, handler, req_class, resp_class):
+            e.message,
+        )
         return Response(
             content=json.dumps({"code": e.code, "message": e.message}),
             status_code=status_code,
             media_type="application/json",
             headers={"Connect-Protocol-Version": "1"},
         )
     except Exception as e:
-        logger.exception(f"Error handling unary RPC: {e}")
+        logger.exception(
+            "[envd] unexpected unary RPC error method=%s error=%s", rpc_method, e
```

Ruff (0.15.5)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)